run-tests.py

```diff
@@ -13,6 +14,22 @@ def cleanup(out):
     ret = ret + s
   return ret
 
+
+# Check for required compilers and exit if any are missing
+CLANG_EXE = shutil.which('clang')
+if not CLANG_EXE:
+    sys.exit('Failed to run tests: clang compiler not found')
+
+GCC_EXE = shutil.which('gcc')
+if not GCC_EXE:
+    sys.exit('Failed to run tests: gcc compiler not found')
+
+os.environ["PATH"] = os.getcwd() + os.pathsep + os.environ["PATH"]
+SIMPLECPP_EXE = shutil.which('simplecpp')
```

Comment:
This is bad. You should not manipulate the path (that's what the `cwd` parameter of `subprocess.Popen()` - which is not needed in this case). And `which` is obviously not the way to lookup the executable we just built and exists in CWD.